src/service/FileTransferService.ts

```diff
@@ -42,10 +47,62 @@ export class FileTransferService {
                 }
             }).catch((err) => {
                 if (err.code === UNSUPPORTED_MEDIA_TYPE) {
-                    throw new Error('Unsupported file type')
+                    throw new Error('Unsupported file type');
                 } else {
-                    throw new Error(err.message)
+                    throw new Error(err.message);
                 }
             });
     }
+
+    async fileMetada(fileId: string): Promise<FileMetada> {
+        const config: AxiosRequestConfig = {
+            headers: {
+                'x-api-key': this.key
+            },
+        };
+
+        return await axios
+            .get<FileMetada>(`${this.url}/${fileId}`, config)
+            .then(_ => _.data)
+            .catch(_ => {
+                throw new FileNotFoundError(fileId, _.message);
+            });
+    }
+
+    async download(fileId: string, res: stream.Writable): Promise<void> {
+
+        const url = `${this.url}/${fileId}/download`;
+        const config: AxiosRequestConfig = {
+            headers: {
+                'x-api-key': this.key
+            },
+            responseType: 'stream'
+        };
+
+        return axios.get<stream.Readable>(url, config)
+            .then(async (axiosResponse: AxiosResponse<stream.Readable>) => {
+
+                if ((res as Response).setHeader) {
```

Comment:
When I saw method signature taking writable stream I thought it is OK but service should know be independent from express.js and as such should know nothing about response.

Please either return readable stream from `download` function or set response headers outside of the download method. This service conceptually close to client library. Setting response headers should be on the controller side.

Reply:
That sounds good to me